src/Hamiltonians/TwoBodyRealCorrelation.jl

```diff
@@ -0,0 +1,63 @@
+"""
+    G2RealCorrelator(d::Int) <: AbstractHamiltonian{Float64}
+
+Two-body operator for density-density correlation between sites separated by `d`
+with `0 ≤ d < M`.
+```math
+    \\hat{G}^{(2)}(d) = \\sum_i^M \\hat{n}_i (\\hat{n}_{i+d} - \\delta_{0d}).
+```
+Can be applied to any [`SingleComponentFockAddress`](@ref).
+Assumes periodic boundary conditions where
+```math
+    \\hat{G}^{(2)}(-M/2 \\leq d < 0) = \\hat{G}^{(2)}(|d|),
+```
+```math
+    \\hat{G}^{(2)}(M/2 < d < M) = \\hat{G}^{(2)}(M - d),
+```
+and normalisation
+```math
+    \\sum_{d=0}^{M-1} \\hat{G}^{(2)}(d) = N (N-1).
+```
+
+# Arguments
+- `d::Integer`: distance between sites.
+
+# See also
+
+* [`HubbardReal1D`](@ref)
+* [`G2Correlator`](@ref)
```

Comment:
Since we now have two G2 correlators, should we rename the old one to `G2MomCorrelator` or something like that?